Modules/mathmodule.c

```diff
@@ -2016,6 +2016,71 @@ math_factorial(PyObject *module, PyObject *arg)
 }
 
 
+/*[clinic input]
+math.lcm
+
+    x as a: object
+    y as b: object
+    /
+least common multiple of x and y
+[clinic start generated code]*/
+
+static PyObject *
+math_lcm_impl(PyObject *module, PyObject *a, PyObject *b)
+/*[clinic end generated code: output=6f83fb6d671074ba input=bd41b785dc2a4ff1]*/
+{
+    PyObject *g, *m, *f, *ab;
+    
+    a = PyNumber_Index(a);
+        if (a == NULL) {
+            return NULL;
+    }
+    b = PyNumber_Index(b);
+    if (b == NULL) {
+        return NULL;
+    }
+    if (a == 0 && b == 0) {
+        return 0;
+    }
+    else {
+        if (PyFloat_Check(a) || PyFloat_Check(b)) {
+            PyErr_Format(PyExc_TypeError,"'float' object cannot be an argument to lcm");
+            return 0;  
+        }
+        g = _PyLong_GCD(a, b);
+        Py_DECREF(a);
```

Comment:
This is the second refcounting bug: at this point we own one reference to `a`, but you're freeing it. That's bad, because we still need to use `a` below. You want to keep ownership of the reference until you no longer need it. You want to get rid of this line.